beep/structure.py

```diff
@@ -168,8 +169,13 @@ def get_interpolated_steps(self, v_range, resolution, step_type='discharge', reg
             new_df = self.data.loc[self.data["cycle_index"] == cycle_index].groupby("step_index").filter(step_filter)
             if new_df.size == 0:
                 continue
-            new_df = get_interpolated_data(new_df, "voltage", field_range=v_range,
-                                           columns=incl_columns, resolution=resolution)
+            if axis != 'voltage':
```

Comment:
What are the other possibilites for "axis"? If it's only going to work for a few specific labels, this loop could be "if axis == 'voltage'", "elif axis == 'current'" (for example), and an "else" with NotImplementedError. Or alternately, to be more flexible, you could in line 161 add "axis" to the list, instead of "voltage" explicitly.

Reply:
If only axis='voltage' works, that can also be asserted earlier in the script.

Reply:
Changed this to have explicitly allowed axes for interpolation.